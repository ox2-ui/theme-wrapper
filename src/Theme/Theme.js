import React, { PropTypes } from 'react';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import theme from '@ox2/theme-mui';

const muiTheme = getMuiTheme(theme);


/**
 * Theme Component
 */
const Theme = ({ children }) => (
  <MuiThemeProvider muiTheme={muiTheme}>
    {children}
  </MuiThemeProvider>
);


Theme.propTypes = {
  children: PropTypes.node.isRequired,

};

export default Theme;
