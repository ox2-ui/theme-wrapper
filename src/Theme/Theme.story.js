import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Theme from './Theme';

storiesOf('Theme', module)
  .add('default', () => (
    <Theme>
      <div>Child</div>
    </Theme>
  ));
