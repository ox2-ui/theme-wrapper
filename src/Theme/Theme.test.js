/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Theme from './Theme';

it('renders correctly', () => {
  const wrapper = mount(
    <Theme>
      <div>Child</div>
    </Theme>
  );

  expect(wrapper).toMatchSnapshot();
});
