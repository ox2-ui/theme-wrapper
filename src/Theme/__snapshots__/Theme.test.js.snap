// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Theme>
  <MuiThemeProvider
    muiTheme={
      Object {
        "appBar": Object {
          "color": "hsla(214, 70%, 53%, 1)",
          "height": 64,
          "padding": 24,
          "textColor": "hsla(0, 0%, 35%, 1)",
          "titleFontWeight": 400,
        },
        "avatar": Object {
          "backgroundColor": "rgb(188, 188, 188)",
          "color": "#ffffff",
        },
        "badge": Object {
          "color": "hsla(0, 0%, 35%, 1)",
          "fontWeight": 500,
          "primaryColor": "hsla(214, 70%, 53%, 1)",
          "primaryTextColor": "hsla(0, 0%, 35%, 1)",
          "secondaryColor": "hsla(214, 70%, 53%, 1)",
          "secondaryTextColor": "hsla(0, 0%, 35%, 1)",
          "textColor": "rgba(0, 0, 0, 0.87)",
        },
        "baseTheme": Object {
          "fontFamily": "Roboto, sans-serif",
          "palette": Object {
            "accent1Color": "hsla(214, 70%, 53%, 1)",
            "accent2Color": "#f5f5f5",
            "accent3Color": "#9e9e9e",
            "alternateTextColor": "hsla(0, 0%, 35%, 1)",
            "borderColor": "#e0e0e0",
            "canvasColor": "#ffffff",
            "clockCircleColor": "rgba(0, 0, 0, 0.07)",
            "disabledColor": "rgba(0, 0, 0, 0.3)",
            "pickerHeaderColor": "#00bcd4",
            "primary1Color": "hsla(214, 70%, 53%, 1)",
            "primary2Color": "#0097a7",
            "primary3Color": "hsla(0, 0%, 35%, 1)",
            "secondaryTextColor": "rgba(0, 0, 0, 0.54)",
            "shadowColor": "rgba(0, 0, 0, 1)",
            "textColor": "rgba(0, 0, 0, 0.87)",
          },
          "spacing": Object {
            "desktopDrawerMenuItemHeight": 48,
            "desktopDropDownMenuFontSize": 15,
            "desktopDropDownMenuItemHeight": 32,
            "desktopGutter": 24,
            "desktopGutterLess": 16,
            "desktopGutterMini": 8,
            "desktopGutterMore": 32,
            "desktopKeylineIncrement": 64,
            "desktopSubheaderHeight": 48,
            "desktopToolbarHeight": 56,
            "iconSize": 24,
          },
        },
        "borderRadius": 2,
        "bottomNavigation": Object {
          "backgroundColor": "#ffffff",
          "height": 56,
          "selectedColor": "hsla(214, 70%, 53%, 1)",
          "selectedFontSize": 14,
          "unselectedColor": "rgba(0, 0, 0, 0.54)",
          "unselectedFontSize": 12,
        },
        "button": Object {
          "height": 36,
          "iconButtonSize": 48,
          "minWidth": 88,
        },
        "card": Object {
          "fontWeight": 500,
          "subtitleColor": "rgba(0, 0, 0, 0.54)",
          "titleColor": "rgba(0, 0, 0, 0.87)",
        },
        "cardMedia": Object {
          "color": "rgba(255, 255, 255, 0.87)",
          "overlayContentBackground": "rgba(0, 0, 0, 0.54)",
          "subtitleColor": "rgba(255, 255, 255, 0.54)",
          "titleColor": "rgba(255, 255, 255, 0.87)",
        },
        "cardText": Object {
          "textColor": "rgba(0, 0, 0, 0.87)",
        },
        "checkbox": Object {
          "boxColor": "hsla(0, 0%, 35%, 1)",
          "checkedColor": "hsla(0, 0%, 35%, 1)",
          "disabledColor": "rgba(0, 0, 0, 0.3)",
          "labelColor": "rgba(0, 0, 0, 0.87)",
          "labelDisabledColor": "rgba(0, 0, 0, 0.3)",
          "requiredColor": "hsla(214, 70%, 53%, 1)",
        },
        "chip": Object {
          "backgroundColor": "rgb(224, 224, 224)",
          "deleteIconColor": "rgba(0, 0, 0, 0.26)",
          "fontSize": 14,
          "fontWeight": 400,
          "shadow": "0 1px 6px rgba(0, 0, 0, 0.12),
              0 1px 4px rgba(0, 0, 0, 0.12)",
          "textColor": "rgba(0, 0, 0, 0.87)",
        },
        "datePicker": Object {
          "calendarTextColor": "rgba(0, 0, 0, 0.87)",
          "calendarYearBackgroundColor": "#ffffff",
          "color": "hsla(214, 70%, 53%, 1)",
          "selectColor": "#0097a7",
          "selectTextColor": "hsla(0, 0%, 35%, 1)",
          "textColor": "hsla(0, 0%, 35%, 1)",
        },
        "dialog": Object {
          "bodyColor": "rgba(0, 0, 0, 0.6)",
          "bodyFontSize": 16,
          "titleFontSize": 22,
        },
        "drawer": Object {
          "color": "#ffffff",
          "width": 256,
        },
        "dropDownMenu": Object {
          "accentColor": "#e0e0e0",
        },
        "enhancedButton": Object {
          "tapHighlightColor": "rgba(0, 0, 0, 0)",
        },
        "flatButton": Object {
          "buttonFilterColor": "#999999",
          "color": "rgba(0, 0, 0, 0)",
          "disabledTextColor": "rgba(0, 0, 0, 0.3)",
          "fontSize": 14,
          "fontWeight": 500,
          "primaryTextColor": "hsla(214, 70%, 53%, 1)",
          "secondaryTextColor": "hsla(214, 70%, 53%, 1)",
          "textColor": "rgba(0, 0, 0, 0.87)",
        },
        "floatingActionButton": Object {
          "buttonSize": 56,
          "color": "hsla(214, 70%, 53%, 1)",
          "disabledColor": "rgb(224, 224, 224)",
          "disabledTextColor": "rgba(0, 0, 0, 0.3)",
          "iconColor": "hsla(0, 0%, 35%, 1)",
          "miniSize": 40,
          "secondaryColor": "hsla(214, 70%, 53%, 1)",
          "secondaryIconColor": "hsla(0, 0%, 35%, 1)",
        },
        "fontFamily": "Roboto, sans-serif",
        "gridTile": Object {
          "textColor": "#ffffff",
        },
        "icon": Object {
          "backgroundColor": "hsla(214, 70%, 53%, 1)",
          "color": "#ffffff",
        },
        "inkBar": Object {
          "backgroundColor": "hsla(214, 70%, 53%, 1)",
        },
        "isRtl": false,
        "listItem": Object {
          "leftIconColor": "#757575",
          "nestedLevelDepth": 18,
          "rightIconColor": "#757575",
          "secondaryTextColor": "rgba(0, 0, 0, 0.54)",
        },
        "menu": Object {
          "backgroundColor": "#ffffff",
          "containerBackgroundColor": "#ffffff",
        },
        "menuItem": Object {
          "dataHeight": 32,
          "height": 48,
          "hoverColor": "rgba(0, 0, 0, 0.1)",
          "padding": 24,
          "rightIconDesktopFill": "#757575",
          "selectedTextColor": "hsla(214, 70%, 53%, 1)",
        },
        "menuSubheader": Object {
          "borderColor": "#e0e0e0",
          "padding": 24,
          "textColor": "hsla(214, 70%, 53%, 1)",
        },
        "overlay": Object {
          "backgroundColor": "rgba(0, 0, 0, 0.54)",
        },
        "palette": Object {
          "accent1Color": "hsla(214, 70%, 53%, 1)",
          "accent2Color": "#f5f5f5",
          "accent3Color": "#9e9e9e",
          "alternateTextColor": "hsla(0, 0%, 35%, 1)",
          "borderColor": "#e0e0e0",
          "canvasColor": "#ffffff",
          "clockCircleColor": "rgba(0, 0, 0, 0.07)",
          "disabledColor": "rgba(0, 0, 0, 0.3)",
          "pickerHeaderColor": "#00bcd4",
          "primary1Color": "hsla(214, 70%, 53%, 1)",
          "primary2Color": "#0097a7",
          "primary3Color": "hsla(0, 0%, 35%, 1)",
          "secondaryTextColor": "rgba(0, 0, 0, 0.54)",
          "shadowColor": "rgba(0, 0, 0, 1)",
          "textColor": "rgba(0, 0, 0, 0.87)",
        },
        "paper": Object {
          "backgroundColor": "#ffffff",
          "color": "rgba(0, 0, 0, 0.87)",
          "zDepthShadows": Array [
            "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
            "0 3px 10px rgba(0, 0, 0, 0.16),
               0 3px 10px rgba(0, 0, 0, 0.23)",
            "0 10px 30px rgba(0, 0, 0, 0.19),
               0 6px 10px rgba(0, 0, 0, 0.23)",
            "0 14px 45px rgba(0, 0, 0, 0.25),
               0 10px 18px rgba(0, 0, 0, 0.22)",
            "0 19px 60px rgba(0, 0, 0, 0.3),
               0 15px 20px rgba(0, 0, 0, 0.22)",
          ],
        },
        "prepareStyles": [Function],
        "radioButton": Object {
          "backgroundColor": "hsla(0, 0%, 35%, 1)",
          "borderColor": "rgba(0, 0, 0, 0.87)",
          "checkedColor": "hsla(214, 70%, 53%, 1)",
          "disabledColor": "rgba(0, 0, 0, 0.3)",
          "labelColor": "rgba(0, 0, 0, 0.87)",
          "labelDisabledColor": "rgba(0, 0, 0, 0.3)",
          "requiredColor": "hsla(214, 70%, 53%, 1)",
          "size": 24,
        },
        "raisedButton": Object {
          "color": "hsla(0, 0%, 35%, 1)",
          "disabledColor": "hsla(0, 0%, 31.5%, 1)",
          "disabledTextColor": "rgba(0, 0, 0, 0.3)",
          "fontSize": 14,
          "fontWeight": 500,
          "primaryColor": "hsla(214, 70%, 53%, 1)",
          "primaryTextColor": "hsla(0, 0%, 35%, 1)",
          "secondaryColor": "hsla(214, 70%, 53%, 1)",
          "secondaryTextColor": "hsla(0, 0%, 35%, 1)",
          "textColor": "rgba(0, 0, 0, 0.87)",
        },
        "rawTheme": Object {
          "fontFamily": "Roboto, sans-serif",
          "palette": Object {
            "accent1Color": "hsla(214, 70%, 53%, 1)",
            "accent2Color": "#f5f5f5",
            "accent3Color": "#9e9e9e",
            "alternateTextColor": "hsla(0, 0%, 35%, 1)",
            "borderColor": "#e0e0e0",
            "canvasColor": "#ffffff",
            "clockCircleColor": "rgba(0, 0, 0, 0.07)",
            "disabledColor": "rgba(0, 0, 0, 0.3)",
            "pickerHeaderColor": "#00bcd4",
            "primary1Color": "hsla(214, 70%, 53%, 1)",
            "primary2Color": "#0097a7",
            "primary3Color": "hsla(0, 0%, 35%, 1)",
            "secondaryTextColor": "rgba(0, 0, 0, 0.54)",
            "shadowColor": "rgba(0, 0, 0, 1)",
            "textColor": "rgba(0, 0, 0, 0.87)",
          },
          "spacing": Object {
            "desktopDrawerMenuItemHeight": 48,
            "desktopDropDownMenuFontSize": 15,
            "desktopDropDownMenuItemHeight": 32,
            "desktopGutter": 24,
            "desktopGutterLess": 16,
            "desktopGutterMini": 8,
            "desktopGutterMore": 32,
            "desktopKeylineIncrement": 64,
            "desktopSubheaderHeight": 48,
            "desktopToolbarHeight": 56,
            "iconSize": 24,
          },
        },
        "refreshIndicator": Object {
          "loadingStrokeColor": "hsla(214, 70%, 53%, 1)",
          "strokeColor": "#e0e0e0",
        },
        "ripple": Object {
          "color": "rgba(0, 0, 0, 0.87)",
        },
        "slider": Object {
          "handleColorZero": "hsla(0, 0%, 35%, 1)",
          "handleFillColor": "hsla(0, 0%, 35%, 1)",
          "handleSize": 12,
          "handleSizeActive": 18,
          "handleSizeDisabled": 8,
          "rippleColor": "hsla(214, 70%, 53%, 1)",
          "selectionColor": "hsla(214, 70%, 53%, 1)",
          "trackColor": "hsla(0, 0%, 35%, 1)",
          "trackColorSelected": "#9e9e9e",
          "trackSize": 2,
        },
        "snackbar": Object {
          "actionColor": "hsla(214, 70%, 53%, 1)",
          "backgroundColor": "rgba(0, 0, 0, 0.87)",
          "textColor": "hsla(0, 0%, 35%, 1)",
        },
        "spacing": Object {
          "desktopDrawerMenuItemHeight": 48,
          "desktopDropDownMenuFontSize": 15,
          "desktopDropDownMenuItemHeight": 32,
          "desktopGutter": 24,
          "desktopGutterLess": 16,
          "desktopGutterMini": 8,
          "desktopGutterMore": 32,
          "desktopKeylineIncrement": 64,
          "desktopSubheaderHeight": 48,
          "desktopToolbarHeight": 56,
          "iconSize": 24,
        },
        "stepper": Object {
          "backgroundColor": "transparent",
          "connectorLineColor": "#bdbdbd",
          "disabledTextColor": "rgba(0, 0, 0, 0.26)",
          "hoverBackgroundColor": "rgba(0, 0, 0, 0.06)",
          "hoveredIconColor": "#616161",
          "iconColor": "hsla(214, 70%, 53%, 1)",
          "inactiveIconColor": "#9e9e9e",
          "textColor": "rgba(0, 0, 0, 0.87)",
        },
        "subheader": Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontWeight": 500,
        },
        "svgIcon": Object {
          "color": "rgba(0, 0, 0, 0.87)",
        },
        "table": Object {
          "backgroundColor": "#ffffff",
        },
        "tableFooter": Object {
          "borderColor": "#e0e0e0",
          "textColor": "#9e9e9e",
        },
        "tableHeader": Object {
          "borderColor": "#e0e0e0",
        },
        "tableHeaderColumn": Object {
          "height": 56,
          "spacing": 24,
          "textColor": "#9e9e9e",
        },
        "tableRow": Object {
          "borderColor": "#e0e0e0",
          "height": 48,
          "hoverColor": "#f5f5f5",
          "selectedColor": "#e0e0e0",
          "stripeColor": "hsla(214, 70%, 76.5%, 0.4)",
          "textColor": "rgba(0, 0, 0, 0.87)",
        },
        "tableRowColumn": Object {
          "height": 48,
          "spacing": 24,
        },
        "tabs": Object {
          "backgroundColor": "hsla(0, 0%, 96%, 1)",
          "selectedTextColor": "hsla(0, 0%, 35%, 1)",
          "textColor": "hsla(0, 0%, 35%, 0.7)",
        },
        "textField": Object {
          "backgroundColor": "transparent",
          "borderColor": "hsla(0, 0%, 86%, 1)",
          "disabledTextColor": "rgba(0, 0, 0, 0.3)",
          "errorColor": "hsla(6, 63%, 46%, 1)",
          "floatingLabelColor": "hsla(0, 0%, 35%, 1)",
          "focusColor": "hsla(214, 70%, 53%, 1)",
          "hintColor": "rgba(0, 0, 0, 0.3)",
          "textColor": "rgba(0, 0, 0, 0.87)",
        },
        "timePicker": Object {
          "accentColor": "hsla(214, 70%, 53%, 1)",
          "clockCircleColor": "rgba(0, 0, 0, 0.07)",
          "clockColor": "rgba(0, 0, 0, 0.87)",
          "color": "hsla(0, 0%, 35%, 1)",
          "headerColor": "#00bcd4",
          "selectColor": "#0097a7",
          "selectTextColor": "hsla(0, 0%, 35%, 1)",
          "textColor": "hsla(0, 0%, 35%, 1)",
        },
        "toggle": Object {
          "labelColor": "hsla(0, 0%, 35%, 1)",
          "labelDisabledColor": "hsla(0, 0%, 50%, 1)",
          "thumbDisabledColor": "#e0e0e0",
          "thumbOffColor": "#f5f5f5",
          "thumbOnColor": "hsla(145, 63%, 42%, 1)",
          "thumbRequiredColor": "hsla(214, 70%, 53%, 1)",
          "trackDisabledColor": "hsla(0, 0%, 86%, 1)",
          "trackOffColor": "hsla(0, 0%, 50%, 1)",
          "trackOnColor": "hsla(214, 70%, 53%, 0.5)",
          "trackRequiredColor": "hsla(214, 70%, 53%, 0.5)",
        },
        "toolbar": Object {
          "backgroundColor": "rgb(232, 232, 232)",
          "color": "rgba(0, 0, 0, 0.54)",
          "height": 56,
          "hoverColor": "rgba(0, 0, 0, 0.87)",
          "iconColor": "rgba(0, 0, 0, 0.4)",
          "menuHoverColor": "rgba(0, 0, 0, 0.1)",
          "separatorColor": "rgba(0, 0, 0, 0.175)",
          "titleFontSize": 20,
        },
        "tooltip": Object {
          "color": "#ffffff",
          "rippleBackgroundColor": "#616161",
        },
        "zIndex": Object {
          "appBar": 1100,
          "dialog": 1500,
          "dialogOverlay": 1400,
          "drawer": 1300,
          "drawerOverlay": 1200,
          "layer": 2000,
          "menu": 1000,
          "popover": 2100,
          "snackbar": 2900,
          "tooltip": 3000,
        },
      }
    }
  >
    <div>
      Child
    </div>
  </MuiThemeProvider>
</Theme>
`;
